src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx

```diff
@@ -0,0 +1,234 @@
+      return getEnvironmentListPromise({
+        queryParams: { ...queryParams, size: options.limit, page: options.offset }
+      })
+    },
+    limit: pageSize.current,
+    loadMoreRef,
+    searchTerm
+  })
+
+  const isFetchingEnvironmentsFirstTime = useMemo(() => {
+    return fetchingEnvironments && offsetToFetch.current === 0
+  }, [fetchingEnvironments, offsetToFetch.current])
+
+  const isFetchingEnvironmentsNextTime = useMemo(() => {
+    return fetchingEnvironments && offsetToFetch.current > 0
+  }, [fetchingEnvironments, offsetToFetch.current])
+
+  const isEmptyContent = useMemo(() => {
+    return !fetchingEnvironments && !fetchEnvironmentsError && isEmpty(environments)
+  }, [fetchingEnvironments, fetchEnvironmentsError, environments])
+
+  return (
+    <Layout.Vertical>
+      <Container margin={{ bottom: 'small' }}>
+        <ExpandingSearchInput
+          alwaysExpanded
+          placeholder={'Search Environments'}
+          autoFocus={false}
+          width={'100%'}
+          onChange={setSearchTerm}
+          throttle={200}
+        />
+      </Container>
+
+      <Layout.Horizontal
+        height={360}
+        border={{
+          radius: 4
+        }}
+      >
+        <Layout.Vertical className={css.editModalSelection} width={'50%'} border={{ right: true }}>
+          {isFetchingEnvironmentsFirstTime ? (
+            <Container flex={{ align: 'center-center' }} height={'100%'}>
+              <Spinner size={32} />
+            </Container>
+          ) : isEmptyContent ? (
+            <Layout.Vertical flex={{ align: 'center-center' }} width={'100%'} height={'100%'}>
+              <img src={EmptyEnvironmentGroup} alt={getString('cd.noEnvironment.title')} />
+              <Heading level={2}>{getString('cd.noEnvironment.title')}</Heading>
+            </Layout.Vertical>
+          ) : (
+            (environments as EnvironmentResponse[]).map((item, index) => {
+              if (!item?.environment) {
+                return null
+              }
+
+              const { name, identifier, tags } = item?.environment
+              const checked = newEnvs.some(
+                (selectedEnv: EnvironmentResponse) => selectedEnv.environment?.identifier === identifier
+              )
+
+              return (
+                <Layout.Horizontal
+                  width={'100%'}
+                  height={60}
+                  flex={{ justifyContent: 'flex-start', alignItems: 'center' }}
+                  padding={{ left: 'medium' }}
+                  margin={{ top: 0 }}
+                  border={{ bottom: true }}
+                  key={identifier}
+                  ref={attachRefToLastElement(index) ? loadMoreRef : undefined}
+                >
+                  <EnvironmentSelection
+                    identifier={identifier}
+                    checked={checked}
+                    onChange={(e: FormEvent<HTMLInputElement>) => {
+                      if ((e.target as any).checked && identifier) {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
                      if ((e.target as unknown).checked && identifier) {
```